src/agoradatatools/etl/transform/immunohisto_transform.py

```diff
@@ -259,28 +267,44 @@ def immunohisto_transform(
     Args:
         datasets (Dict[str, pd.DataFrame]): Dictionary of dataset names mapped to their DataFrame.
         dataset_name (str): The name of the dataset to transform.
+        required_input (Dict[str, List[str]], optional): Required input columns for validation.
         group_columns (List[str], optional): List of columns to group by. Defaults to ['name', 'evidence_type', 'tissue', 'age', 'units'].
         extra_columns (List[str], optional): List of columns to include in the group. Defaults to ['genotype', 'sex', 'individual_id', 'value'].
         extra_column_name (str, optional): Name of the column containing the extra columns. Defaults to 'data'.
 
     Returns:
-        pd.DataFrame: DataFrame containing all group_columns, plus an extra column named as specified
+        List[Dict[str, Any]]: List of dictionaries containing all group_columns, plus an extra column named as specified
         in extra_column_name. This extra column contains all information from extra_columns, collapsed
         into a single dictionary.
     """
 
-    # Filter required_input to only include datasets that are present
-    filtered_required_input = {
-        key: value for key, value in required_input.items() if key in datasets
-    }
-
+    # Validate that required datasets are present
     # Ensure at least one of "biomarkers" or "pathology" is present
     if not any(key in datasets for key in ["biomarkers", "pathology"]):
         raise ValueError(
             "At least one of 'biomarkers' or 'pathology' must be present in the datasets"
         )
 
-    check_required_datasets_and_columns(datasets, filtered_required_input)
+    # Convert immunohisto_measure_order DataFrame format if needed
+    # Handle generic column names from read_yaml_into_df (key/items) or legacy names (dataset_name/evidence_type)
+    if "immunohisto_measure_order" in datasets:
+        config_df = datasets["immunohisto_measure_order"].copy()
```

Comment:
I just tested this: Instead of having this rename code in several places, you can delete it and just add the following to the `column_rename` section of the config for each thing that uses immunohisto_measure_order:
```
        key: dataset_name
        items: evidence_type
```
The dataframe comes into the transform with the correct names.